atomic_reactor/plugins/post_compare_components.py

```diff
@@ -0,0 +1,81 @@
+"""
+Copyright (c) 2017 Red Hat, Inc
+All rights reserved.
+
+This software may be modified and distributed under the terms
+of the BSD license. See the LICENSE file for details.
+"""
+from atomic_reactor.plugin import PostBuildPlugin, PluginFailedException
+from atomic_reactor.constants import (PLUGIN_COMPARE_COMPONENTS_KEY,
+                                      PLUGIN_FETCH_WORKER_METADATA_KEY)
+
+
+class CompareComponentsPlugin(PostBuildPlugin):
+    """
+    Compare components from each worker build and verify the same version was
+    on each worker.
+    """
+
+    key = PLUGIN_COMPARE_COMPONENTS_KEY
+    is_allowed_to_fail = False
+
+    def rpm_compare(self, a, b):
+        """
+        Compare rpm component version 'a' with 'b'.
+        'name' is implied equal as it is the key() lookup
+        """
+        if a['version'] == b['version'] and \
+           a['release'] == b['release'] and \
+           a['signature'] == b['signature']:
```

Comment:
For completeness, let's compare the `'type'` key as well.

Reply:
(This looks fine now we're using (type,name) as the identifier.)